pkg/storage/storagewrappers/cached_datastore.go

```diff
@@ -394,6 +400,7 @@ func (c *CachedDatastore) newCachedIterator(
 		span.SetAttributes(attribute.Bool("cached", true))
 
 		staticIter := storage.NewStaticIterator[*storage.TupleRecord](cacheEntry.Tuples)
+		currentIteratorCacheCount.Inc()
```

Comment:
@adriantam wouldn't this one never be decreased?

Reply:
It is decremented at https://github.com/openfga/openfga/blob/d737c55a5287cc986b31f8e9bdda7ed49e3e7f76/pkg/storage/storagewrappers/cached_datastore.go#L526

Note that both the instance generated from cache and not from cache are "separate" entity and require .Stop() to be called separately.

Reply:
note this is an instance of `cachedTupleIterator`, not `cachedIterator`